rhel/repo2cpe/updater.go

```diff
@@ -0,0 +1,164 @@
+package repo2cpe
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"time"
+
+	"github.com/rs/zerolog"
+)
+
+type LocalFileMetadata struct {
+	lastUpdateDate  time.Time
+	lastHeaderQuery time.Time
+}
+
+// LocalUpdaterJob periodically updates mapping file and store it in local storage
+type LocalUpdaterJob struct {
+	URL             string
+	Mapping         MappingFile
+	Client          *http.Client
+	lastUpdateDate  time.Time
+	lastHeaderQuery time.Time
+}
+
+// NewLocalUpdaterJob creates new LocalUpdaterJob
+func NewLocalUpdaterJob(url string, client *http.Client) *LocalUpdaterJob {
```

Comment:
I am not seeing where this becomes "long-running" or triggered on an interval.

Reply:
`Update` function is called in `RepositoryToCPE` and it updates mapping if needed (based on Last-modified http flag)

Reply:
@allda

When Clair is in the process of matching, its going to call more then just RHEL matcher concurrently. 

If RHEL matcher cannot reach mapping file, because maybe the server is down, the RHEL marcher should not return an error, if it does the entire match process will fail. 

Can you confirm that, if RHEL matcher needs to update the mapping file, but cannot reach the endpoint where it exists, RHEL matcher will gracefully return no results and not return an Error?